bin/hdfcoinc/pycbc_multiifo_combine_statmap

```diff
@@ -0,0 +1,182 @@
+    f[group_to_save + '/start'] = fi[ifo_to_get + '/start'][:]
+    #also move the attributes with the given ifo combination indicator
+    f.attrs['pivot/{}'.format(ifosstring)] = fi.attrs['pivot']
+    f.attrs['fixed/{}'.format(ifosstring)] = fi.attrs['fixed']
+    f.attrs['timeslide_interval/{}'.format(ifosstring)] = fi.attrs['timeslide_interval']
+    f.attrs['background_time/{}'.format(ifosstring)] = fi.attrs['background_time']
+    f.attrs['foreground_time/{}'.format(ifosstring)] = fi.attrs['foreground_time']
+    f.attrs['background_time_exc/{}'.format(ifosstring)] = fi.attrs['background_time_exc']
+    f.attrs['foreground_time_exc/{}'.format(ifosstring)] = fi.attrs['foreground_time_exc']
+
+    
+
+key_set = pycbc.io.name_all_datasets(files)
+# copy and concatenate all the columns in the foreground group 
+# from all files /except/ the IFO groups, we recalculate ifar/fap later
+for key in key_set:
+    if key.startswith('foreground') and not any([ifo in key for ifo in args.ifos]):
+        pycbc.io.combine_and_copy(f, files, key)
+
+
+logging.info('collating triggers into single structure')
+#for the ifo-specific groups, copy them over into the new file
+# - if the ifo is not present for each coincident trigger, 
+#  indicate by including a '-1' value in the combined file
+
+trigger_list = []
+all_trig_times = {}
+all_trig_ids = {}
+for ifo in args.ifos:
+    all_trig_times[ifo] = numpy.array([], dtype=numpy.uint32)
+    all_trig_ids[ifo] = numpy.array([], dtype=numpy.uint32)
+
+# for each file, append the trigger time and id data for each ifo
+#   - if it isnt there, then fill with -1 values
+for f_in in files:
+    for ifo in args.ifos:
+        if ifo in f_in['foreground']:
+            all_trig_times[ifo] = numpy.concatenate([all_trig_times[ifo], \
+                                     f_in['foreground/{}/time'.format(ifo)][:]])
+            all_trig_ids[ifo] = numpy.concatenate([all_trig_ids[ifo], 
+                                     f_in['foreground/{}/trigger_id'.format(ifo)][:]])
+        else:
+            all_trig_times[ifo] = numpy.concatenate([all_trig_times[ifo],
+                                     -1*numpy.ones_like(f_in['foreground/fap'][:], dtype=numpy.uint32)])
+            all_trig_ids[ifo] = numpy.concatenate([all_trig_ids[ifo],
+                                     -1*numpy.ones_like(f_in['foreground/fap'][:], dtype=numpy.uint32)])
+
+logging.info('clustering triggers for loudest stat value')
+#cluster the triggers before applying trials factor
+for ifo in args.ifos:
+    f['foreground/{}/time'.format(ifo)] = all_trig_times[ifo]
+    f['foreground/{}/trigger_id'.format(ifo)] = all_trig_ids[ifo]
+
+#cluster for loudest stat value 
+def argmax(v):
+    return numpy.argsort(v)[-1]
+
+stat = numpy.core.records.fromarrays([f['foreground/ifar'][:],
+                                      f['foreground/stat'][:]],
+                                      names='ifar,stat')
+
```

Comment:
This is inherited from previous code, but based on https://docs.scipy.org/doc/numpy/reference/generated/numpy.core.records.fromarrays.html, it will stay in the same order

Storing it like this means that you can use the numpy.argsort to break degeneracies in ifar

Or have I missed the point of the question?

Reply:
we worked out that all relevant arrays were sorted first by order of input files, then by order of ifo as given in the command line argument, so this is not an issue.